rust/sdk/examples/chat/display.rs

```diff
@@ -127,6 +83,86 @@ pub async fn display(mut cell: ActoCell<Display, impl ActoRuntime>) {
     }
 }
 
+fn render<W: io::Write>(
+    f: &mut Frame<CrosstermBackend<W>>,
+    identity: &Reader<Identity>,
+    text_area: &Reader<TextArea>,
+    messages: &Reader<Vec<Message>>,
+    not_connected: &Option<String>,
+) {
+    identity.project(|identity| match &identity.edit {
+        Some(edit) => render_editing_identity(f, edit),
+        None => render_chat(f, text_area, messages, not_connected),
```

Comment:
Remember that `identity.project()` holds a lock. Calling other functions that may obtain locks is best avoided, otherwise you get to analyse call trees and prove that no deadlocks can occur :-)

This can be fixed by returning `bool` from the projection function to indicate whether `render_chat` should still be called.

Reply:
Ah yes. For the sake of future programmers who don't have as much context as I do at the time of writing, that will be better.